### DISCUSSION TITLE
Unable to Exit Full-Screen Mode in FreeRDP 3.8.0 on Linux – Any Workarounds?

### DISCUSSION THREAD

**Discussion by kaiserfr** [FreeRDP/FreeRDP#10583] [Q&A]
Hi everyone,

I'm encountering an issue with FreeRDP where I can't exit full-screen mode once it's started using the `/f` option. I know this issue has already been reported in [Issue #9886](https://github.com/FreeRDP/FreeRDP/issues/9886) (open since February), but I'm working with a Linux host system instead of the one mentioned there.

Has anyone found a workaround to regain access to the host system while FreeRDP is running in full-screen mode? For example, is there a way to grab keystrokes before FreeRDP captures them all?

Any help would be greatly appreciated!

**Comment by akallabeth** [comment]
@kaiserfr what are you talking about? The referenced issue is for the native mac client, so I doubt you are using that one ;)
as for your issue, it might help if we knew which client you are using:
keyboard shortcuts:
* `xfreerdp` -> `<ctrl>+<alt>+<return>` 
* `sdl-freerdp` -> `<right shift>+<return>` (modifiable via config file)

**Reply by kaiserfr** [reply]
Yes, you're right, the mentioned issue is specific to the native mac client, but the underlying problem seems to be similar.

I'm currently using FreeRDP 3.8.0 (Flatpak build) on a Linux Mint 22 Wilma system with a GNOME desktop environment.

I've already tried the ctrl + alt + return shortcut, but it doesn't work for me.

The combination right shift + return is new to me - it’s not listed in the command-line help. When I tried it, I got a partial reaction: the Windows 11 desktop inside FreeRDP went black, but it didn’t go to the background. As I move the mouse, some elements below the mouse trace reappear.

**Reply by akallabeth** [reply]
@kaiserfr it is listed in the command line help for `sdl-freerdp`:
```
CONFIGURATION FILE

  The SDL client supports some user defined configuration options.
  Settings are stored in JSON format
  The location is a per user file. Location for current user is /home/nin/.var/app/com.freerdp.FreeRDP/config/freerdp/sdl-freerdp.json
  The XDG_CONFIG_HOME environment variable can be used to override the base directory.

  The following configuration options are supported:

    SDL_KeyModMask
      Defines the key combination required for SDL client shortcuts.
      Default KMOD_RSHIFT
      An array of SDL_Keymod strings as defined at https://wiki.libsdl.org/SDL2/SDL_Keymod

    SDL_Fullscreen
      Toggles client fullscreen state.
      Default SDL_SCANCODE_RETURN.
      A string as defined at https://wiki.libsdl.org/SDL2/SDLScancodeLookup

    SDL_Resizeable
      Toggles local window resizeable state.
      Default SDL_SCANCODE_R.
      A string as defined at https://wiki.libsdl.org/SDL2/SDLScancodeLookup

    SDL_Grab
      Toggles keyboard and mouse grab state.
      Default SDL_SCANCODE_G.
      A string as defined at https://wiki.libsdl.org/SDL2/SDLScancodeLookup

    SDL_Disconnect
      Disconnects from the RDP session.
      Default SDL_SCANCODE_D.
      A string as defined at https://wiki.libsdl.org/SDL2/SDLScancodeLookup
```

**Reply by kaiserfr** [reply]
You are referring to SDL_SCANCODE_RETURN ?
How would I know that SCANCODE == right shift ?

Anyway - although right shift + return triggers a reaction I'm still not able to return to my host computer.
I also got no reaction from SDL_Resizeable and SDL_Grab. SDL_Disconnect however works.

(Edit: I now understand that the right shift key is defined in SDL_KeyModMask. 
But please simplify the commandline-help so that the casual user can understand it too)

**Reply by akallabeth** [reply]
@kaiserfr from the name or the [link](https://wiki.libsdl.org/SDL2/SDLScancodeLookup) directly below?
anyway, pull requests are very welcome ;)

**Reply by akallabeth** [reply]
@kaiserfr just tested, `sdl-freerdp` perfectly fine leaves fullscreen mode (you can then `<alt>+<tab>` on host) but does not resize the window (more or less expected) - so that might be confusing.
anyway, created https://github.com/FreeRDP/FreeRDP/pull/10584 to add a minimize shortcut as that is most likely the key combo that most want (minimize to tray, do something, restore with `<alt>+<tab>` without leaving fullscreen)

as for documentation, if you find a better text feel free to create a pull request / an issue with the changes, a patch set or whatever, but only have limited time and this is quite time consuming so any help appreciated.

**Reply by kaiserfr** [reply]
I tested with first `<rightshift>+<enter>` followed by `<alt>+<tab>` as you proposed and it still does not work for me.
Appreciate opening #10584 as this may open up another way to toggle between freerdp fullscreen and host.

As for writing documentation I will consider helping out.

**Comment by kaiserfr** [comment]
Finally it worked for me with `<right-shift>-<return>` and when I open my connection with the parameters `/multimon -decorations`, which opens two full-size screens. With other parameters, e.g. `/f` on one screen it did not work.

**Comment by ghost** [comment]
Summary for sdl-freerdp3:

Toggle grab with `<right-shift>+G`. After that you can use the normal shortcuts from your window manager.     
This shortcut can be changed. See `man sdl-freerdp3` and search for 'configuration file'
